services/shhext/api.go

```diff
@@ -180,6 +180,60 @@ func (api *PublicAPI) getPeer(rawurl string) (*enode.Node, error) {
 	return enode.ParseV4(rawurl)
 }
 
+// RetryConf specifies configuration for retries with timeout and max amount of retries.
+type RetryConf struct {
+	BaseTimeout time.Duration
+	// StepTimeout defines duration increase per each retry.
+	StepTimeout time.Duration
+	MaxRetries  int
+}
+
+// RequestMessagesSync repeats MessagesRequest using configuration in retry conf.
+func (api *PublicAPI) RequestMessagesSync(conf RetryConf, r MessagesRequest) error {
+	shh := api.service.w
+	events := make(chan whisper.EnvelopeEvent, 10)
+	sub := shh.SubscribeEnvelopeEvents(events)
+	defer sub.Unsubscribe()
+	var (
+		hid     hexutil.Bytes
```

Comment:
What does `hid` stand for?

Reply:
hash id :) probably better name is requestID